tests/avalon_test_framework/conftest.py

```diff
@@ -31,3 +31,46 @@ def setup_config(args=None):
                                            logging.WARN))
 
     logger.info("configuration for the session: %s", config)
+
+
+@pytest.fixture(scope="function")
+def setup_teardown(request):
+    marker = request.node.get_closest_marker("setup_teardown_data")
+    method_name = None
+    if marker is None:
+        # Handle missing marker in some way...
+        test_obj = None
+        exit(0)
+    else:
+        use_fixture = True
+        test_obj = marker.args[0]
+        method_name = marker.args[1]
+        if len(marker.args) == 3:
+            use_fixture = marker.args[2]
+        if use_fixture:
+            pre_test_output = None
+            if "WorkerLookUp" in method_name or \
+                    "WorkerRegister" in method_name:
```

Comment:
Update this to method_name in ["WorkerLookUp", "WorkerRegister"]

Reply:
I am eventually planning to have couple of lists defined in the new toml file that Manoj is working on (pre_worker_env and pre_workorder_env) where it will all the methods that should run the corresponding setup and just say if method_name in those lists. So have not optimized this part.